src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs

```diff
@@ -32,23 +32,52 @@ public unsafe static NetworkInterface[] GetLinuxNetworkInterfaces()
             const int MaxTries = 3;
             for (int attempt = 0; attempt < MaxTries; attempt++)
             {
+                // Because these callbacks are executed in a reverse-PInvoke, we do not want any exceptions
+                // to propogate out, because they will not be catchable. Instead, we track all the exceptions
+                // that are thrown in these callbacks, and aggregate them at the end.
+                List<Exception> exceptions = new List<Exception>();
```

Comment:
Also, maybe move the List declaration up to the same place as the Dictionary (and clear it the same place the dictionary is cleared, if it isn't null), so it isn't allocated on each retry iteration?